larray/core/group.py

```diff
@@ -1275,6 +1276,74 @@ def containing(self, substring):
             substring = substring.eval()
         return LGroup([v for v in self.eval() if substring in v], axis=self.axis)
 
+    def to_hdf(self, filepath, key=None, axis_key=None, **kwargs):
+        """
+        Writes group to a HDF file.
+
+        A HDF file can contain multiple groups.
+        The 'key' parameter is a unique identifier for the group.
+        The 'axis_key' parameter is the unique identifier for the associated axis.
+        The associated axis will be saved if not already present in the HDF file.
+
+        Parameters
+        ----------
+        filepath : str
+            Path where the hdf file has to be written.
+        key : str or Group, optional
+            Name of the group within the HDF file.
+            If None, the name of the group is used.
+            Defaults to None.
+        axis_key : str, optional
+            Key of the associated axis in the HDF file.
+            If None, the name of the axis associated with the group is used.
+            Defaults to None.
+        **kwargs
+
+        Examples
+        --------
+        >>> from larray import Axis
+        >>> a = Axis("a=a0..a2")
+        >>> a.to_hdf('test.h5')
+        >>> a01 = a['a0,a1'] >> 'a01'
+
+        Save group
+
+        >>> # by default, the key is the name of the group
+        >>> # and axis_key the name of the associated axis
+        >>> a01.to_hdf('test.h5')                  # doctest: +SKIP
+
+        Save group with a specific key
+
+        >>> a01.to_hdf('test.h5', 'a_01')           # doctest: +SKIP
+
+        Save group in a specific HDF group
+
+        >>> a.to_hdf('test.h5', 'groups/a01')       # doctest: +SKIP
+
+        The associated axis is saved with the group if not already present in the HDF file
+
+        >>> b = Axis("b=b0..b2")
+        >>> b01 = b['b0,b1'] >> 'b01'
+        >>> # save both the group 'b01' and the associated axis 'b'
+        >>> b01.to_hdf('test.h5')                   # doctest: +SKIP
+        """
+        if key is None:
+            if self.name is None:
+                raise ValueError("Argument key must be provided explicitly in case of anonymous group")
+            key = self.name
+        key = _translate_key_hdf(key)
```

Comment:
only translate if key comes from self.name?

Reply:
Isn't the opposite? Only translate if key does not come from self.name?

Reply:
Oh. My bad, sorry. I thought it translated "raw" strings too. In this case, leave this line as-is, it's fine.

That's the problem with reviews on github, you don't have goto definition. Changing the method name to something like _translate_group_key_hdf to avoid such confusion might be a good idea (or not 😉).